package.json

```
@@ -1,47 +1,61 @@
 {
   "name": "serverless-log-forwarding",
-  "version": "2.0.0",
+  "version": "3.0.0",
   "description": "a serverless plugin to forward logs to given lambda function",
-  "main": "index.js",
+  "keywords": [
+    "serverless",
+    "plugin",
+    "logs"
+  ],
+  "homepage": "https://github.com/amplify-education/serverless-log-forwarding#readme",
+  "bugs": {
+    "url": "https://github.com/amplify-education/serverless-log-forwarding/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/amplify-education/serverless-log-forwarding.git"
+  },
+  "license": "MIT",
+  "author": "Amplify Education",
+  "main": "dist/src/index.js",
   "directories": {
     "test": "test"
   },
+  "scripts": {
+    "build": "tsc --project .",
+    "integration-test": "nyc mocha -r ts-node/register --project tsconfig.json test/integration-tests/integration-tests.ts && nyc report --reporter=text-summary",
+    "lint": "eslint . --ext .ts",
+    "test": "nyc mocha -r ts-node/register --project tsconfig.json test/unit-tests/index-test.ts && nyc report --reporter=text-summary"
+  },
   "dependencies": {
-    "underscore": "^1.8.3"
+    "underscore": "^1.11.4"
   },
   "devDependencies": {
+    "@types/async-retry": "^1.3.3",
+    "@types/chai": "^3.5.0",
+    "@types/mocha": "^8.2.0",
+    "@types/randomstring": "^1.1.8",
+    "@types/shelljs": "^0.8.11",
+    "@typescript-eslint/eslint-plugin": "^5.11.0",
+    "@typescript-eslint/parser": "^5.11.0",
+    "async-retry": "^1.3.3",
     "aws-sdk": "^2.1061.0",
     "chai": "^3.5.0",
     "eslint": "^8.7.0",
     "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^16.1.0",
     "eslint-plugin-import": "^2.7.0",
     "eslint-plugin-jsx-a11y": "^6.5.1",
     "eslint-plugin-react": "^7.28.0",
-    "istanbul": "^0.4.5",
     "mocha": "^9.1.4",
+    "nyc": "^15.1.0",
     "randomstring": "^1.2.2",
-    "serverless": "^3.1.0",
-    "shelljs": "^0.8.5"
-  },
-  "scripts": {
-    "test": "node ./node_modules/istanbul/lib/cli.js cover ./node_modules/mocha/bin/_mocha --recusrive ./test/unit-tests/ -- -R spec && node ./node_modules/istanbul/lib/cli.js check-coverage --line 70 coverage/coverage.json",
-    "integration-test": "node ./node_modules/istanbul/lib/cli.js cover ./node_modules/mocha/bin/_mocha --recusrive ./test/integration-tests/ -- -R spec && node ./node_modules/istanbul/lib/cli.js check-coverage --line 70 coverage/coverage.json",
-    "lint": "eslint ."
-  },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/amplify-education/serverless-log-forwarding.git"
-  },
-  "keywords": [
-    "serverless",
-    "serverless",
-    "plugin",
-    "logs"
-  ],
-  "author": "Amplify Education",
-  "license": "MIT",
-  "bugs": {
-    "url": "https://github.com/amplify-education/serverless-log-forwarding/issues"
+    "serverless": "<4",
```

```suggestion
    "serverless": "^3.1.0",
```
Also, you can add 
```
"peerDependencies": {
    "serverless": "<4"
  }
```
To support sls version 1 | 2 | 3

As we did for the Domain Manager https://github.com/amplify-education/serverless-domain-manager/blob/master/package.json